### PR TITLE
test: add unit tests for baked-in template discovery and filtering

### LINKED ISSUE #1850: Add unit tests for baked‑in template discovery/filtering
The baked‑in template helpers in generator/apps/generator/lib/templates/bakedInTemplates.js are core functionality, but there are no dedicated unit tests covering them. This makes it easy to regress template discovery and filtering behavior without noticing.

What to add
- Tests for listBakedInTemplates (no filter, single filter, combined filters)
- Tests for isCoreTemplate (existing + missing names)
- Tests for getTemplate (path resolution + fallback)

### PR DESCRIPTION
**Pull request by miantalha45**
## Description

Adds comprehensive unit tests for the baked-in template discovery and filtering functionality in `lib/templates/bakedInTemplates.js`. This addresses the gap where core template discovery functions had no dedicated test coverage.

## What's Changed

- Added `test/bakedInTemplates.test.js` with 31 test cases covering:
  - `listBakedInTemplates`: 16 tests covering no filter, single filters (type, stack, protocol, target), combined filters, and edge cases
  - `isCoreTemplate`: 7 tests covering existing/non-existing templates and edge cases (null, undefined, empty string, case sensitivity)
  - `getTemplate`: 8 tests covering path resolution, fallback behavior, and error handling

## Testing

All 31 tests pass:
- No filter scenarios
- Single filter scenarios (type, stack, protocol, target)
- Combined filter scenarios
- Edge cases and error handling
- Path resolution and fallback logic

## Related Issue

Fixes #1850 

## Checklist

- [x] Tests added/updated
- [x] All tests pass
- [x] Code follows project style guidelines
- [x] No linting errors

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test coverage for baked-in templates: listing with varied filters (type, protocol, stack, target), core-template checks, and retrieval behaviors including explicit and fallback paths; includes new fixture data covering templates with and without explicit paths.
* **Bug Fixes**
  * Hardened template retrieval: invalid or missing names now return no result; path resolution now reliably prefers explicit template paths when present.

### CODE REVIEW

File: apps/generator/test/bakedInTemplates.test.js [hunk 1]
```diff
@@ -0,0 +1,221 @@
+/**
+ * @jest-environment node
+ */
+const path = require('path');
+const { listBakedInTemplates, isCoreTemplate, getTemplate } = require('../lib/templates/bakedInTemplates');
+const templates = require('../lib/templates/BakedInTemplatesList.json');
```
Comment (derberg) [inline apps/generator/test/bakedInTemplates.test.js]: you cannot test using production data because any changes in the list of templates can cause false/positive errors in tests - you need to create a dummy fixture that will be used only in tests

File: apps/generator/lib/templates/bakedInTemplates.js [hunk 2]
```diff
@@ -43,9 +43,14 @@ module.exports.isCoreTemplate = (templateName) => {
  * @returns {Promise<{name: string, path: string}|undefined>} An object containing the template's name and path, or undefined if not found.
```
[the remark this answers is not in the document]
Comment (derberg) [reply apps/generator/lib/templates/bakedInTemplates.js]: @miantalha45 please look at CodeRabbit, it rarely makes mistakes

File: apps/generator/lib/templates/bakedInTemplates.js [hunk 3]
```diff
@@ -43,9 +43,14 @@ module.exports.isCoreTemplate = (templateName) => {
...
  */
 module.exports.getTemplate = async (templateName) => {
```
Comment (derberg) [inline apps/generator/lib/templates/bakedInTemplates.js:46]: I'm not fan of throwing errors but rather gracefully going forward if possible. So if template name is invalid, it should work like the template name is not on the list, and return empty array, not throw error

**Review by derberg — APPROVED** [review]

**Comment by derberg** [comment]
please fix linter issues

**Comment by derberg** [comment]
do you plan to work on that @miantalha45 or you drop out cause of GSoC not accepting AsyncAPI?

**Comment by miantalha45** [comment]
No no
I am working on it.
